src/Pact/Types/Type.hs

```diff
@@ -299,18 +298,23 @@ canUnifyWith (TyVar SchemaVar {}) (TyVar SchemaVar {}) = True
 canUnifyWith (TyVar (TypeVar _ ac)) (TyVar (TypeVar _ bc)) = all (`elem` ac) bc
 canUnifyWith (TyVar (TypeVar _ cs)) b = null cs || b `elem` cs
 canUnifyWith (TyList a) (TyList b) = a `canUnifyWith` b
-canUnifyWith (TySchema _ aTy aP) (TySchema _ bTy bP) = aTy `canUnifyWith` bTy && aP `canUnifyPartial` bP
+canUnifyWith (TySchema _ aTy aP) (TySchema _ bTy bP)
+  = aTy `canUnifyWith` bTy && aP `canUnifyPartial` bP
 canUnifyWith a b = a == b
 {-# INLINE canUnifyWith #-}
 
--- | a `canUnifyPartial` b means the subset specified by b is
--- valid in a, etc. [] in a subset is a wildcard.
+-- | a `canUnifyPartial` b means the subset specified by b is valid in a, etc.
 canUnifyPartial :: SchemaPartial -> SchemaPartial -> Bool
-canUnifyPartial SPFull SPFull = True
-canUnifyPartial (SPPartial a) (SPPartial b) = b `isSubsetOf` a
-canUnifyPartial SPFull SPPartial {} = True -- full can always satisfy subset
-canUnifyPartial SPPartial {} SPFull = False -- partial cannot satisfy full
-
+canUnifyPartial = flip isSubPartial
+
+-- | @a `isSubPartial` b@ means that @a <= b@ in the lattice given by
+-- @SchemaPartial@, ie, that @a@ is smaller than @b@.
+isSubPartial :: SchemaPartial -> SchemaPartial -> Bool
```

Comment:
It would seem that `isSubPartial` is the more semantically clear so suggest remove `canUnifyPartial` and directly call it above